libs/s25main/ingameWindows/IngameWindow.cpp

```diff
@@ -59,12 +60,17 @@ IngameWindow::IngameWindow(unsigned id, const DrawPoint& pos, const Extent& size
 
 IngameWindow::~IngameWindow()
 {
-    // Possibly save our old position
-    auto settings = SETTINGS.windows.persistentSettings.find(GetGUIID());
-    if(settings != SETTINGS.windows.persistentSettings.end())
+    try
     {
-        settings->second.lastPos = GetPos();
-        settings->second.isOpen = !closeme;
+        // Possibly save our old position
+        auto& settings = SETTINGS.windows.persistentSettings.at(GetGUIID());
+        settings.lastPos = GetPos();
+        settings.isOpen = !closeme;
+    } catch(const std::out_of_range& oor)
+    { // Window is not persistent, no problem
+    } catch(std::runtime_error& err)
+    { // SETTINGS was probably destroyed already, don't save but print a warning
```

Comment:
I added the try catch block because clang-tidy doesn't like it if destructors can throw exceptions. I don't know if it's possible but I guess it doesn't hurt to catch it here

Reply:
I see. However in that case the exception should really be thrown, as it means we need to rethink that case and fully avoid it instead of just logging it (which likely no one will ever notice...).
Thinking about alternatives...
We need to set the position and the open flag. So I guess it should work to set the position in the SetPos function and isOpen in the ctor and reset in Close. Yes it means a few more calls to this code (maybe make it a function?) but I think doing bookkeeping in the destructor is not "clean" as shown by the need to handle this exception.

What do you think?

Reply:
I'm not the biggest fan, because it's easier to forget if someone adds another function that changes those values in the future. But I see your point and don't see a better alternative without using SETTINGS from the destructor, so I just changed it the way you suggested anyway.

Reply:
I know... But it's a good compromise IMO and in theory there should be tests for that new function which should also tests this... But yeah.
Thanks for the changes, LGTM